src/commands/deploy/buildImageInAzure/IBuildImageInAzureContext.ts

```diff
@@ -0,0 +1,23 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { Run as AcrRun, ContainerRegistryManagementClient } from '@azure/arm-containerregistry';
+import * as vscode from 'vscode';
+import { IDeployFromRegistryContext } from '../deployFromRegistry/IDeployFromRegistryContext';
+
+export interface IBuildImageInAzureContext extends IDeployFromRegistryContext {
+    rootFolder: vscode.WorkspaceFolder;
+    dockerFilePath: string;
+    imageName: string;
+    os: string;
```

Comment:
For additional type-safety/clarity, you could make this type:
```suggestion
    os: 'Windows' | 'Linux';
```
Doing this will only allow us to type one or the other, not just any string.